rfc/rfc-0000-maven-artifacts-support.md

```diff
@@ -0,0 +1,130 @@
+# Meta
+[meta]: #meta
+- Name: Maven Artifacts Support
+- Start Date: 2022-03-30
+- Author(s): @emmjohnson
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Modify `owner.spec.source` to allow for specifying a Maven and Artifactory repository.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+  - Today we support workloads being built from git and image source. Many developers in the field also need support
+    Maven and Artifactory repositories hosting their JAR and WAR files.
+- What use cases does it support?
+  - Maven and Artifactory repositories
+- What is the expected outcome?
+  - To enable more types of workloads.
+
+# What it is
+[what-it-is]: #what-it-is
+We need to modify the Owner specs to create an unambiguous way for specifying a JAR file. There are two proposed
+options, the goal is to get community buy-in on a solution before ratifying and the RFC will then be modified to suggest
+the agreed upon spec.
+
+This option allows for the developer to know what is necessary for a maven workload by examining the spec.
+```yaml
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  name: my-workload
+spec:
+  source:
+      maven: 
+        url: https://maven.pkg.jetbrains.space/mycompany/p/projectkey/my-maven-repo
+
+---
+apiVersion: carto.run/v1alpha1
+kind: ClusterSupplyChain
+metadata:
+  name: my-supply-chain
+spec:
+  selectorMatchFields:
+    - key: spec.maven.url
+      operator: Exists
+  resources:
+    - name: source-provider
+      templateRef:
+        name: source-template
+        kind: ClusterSourceTemplate
+```
+
+This option allows us to support many artifacts, but the platform operator would have to communicate to the developer what parameter to use. 
```

Comment:
I favour this second approach. We have been called on to support not just general Maven artifacts but specifically Artifactory artifacts, too. I wouldn't be surprised if there are customer use cases for Nexus artifacts or even things like NPM, too. This second approach seems more scaleable.

Reply:
The reason that we are calling out Artifactory repositories specifically is that customers are asking for Artifactory support explicitly.

Reply:
> customers are asking for Artifactory support explicitly

Let's focus on use cases for Cartographer as an open source project in these spaces 🙂

Reply:
I have removed references to "Artifactory" from my design document.  Our team has decided to focus on Maven functionality first without any Artifactory-specific functionality.  That should suit the open source project, right?

Reply:
Was more calling out the unqualified reference to "customers" -- Artifactory is fine :)